### PR TITLE
pallet-proxy: emit events on proxy added.

### PR DESCRIPTION
**Pull request by shaunxw**

### CODE REVIEW

File: frame/proxy/src/lib.rs [hunk 1]
```diff
@@ -228,7 +228,9 @@ pub mod pallet {
 			delay: T::BlockNumber,
 		) -> DispatchResult {
 			let who = ensure_signed(origin)?;
-			Self::add_proxy_delegate(&who, delegate, proxy_type, delay)
+			Self::add_proxy_delegate(&who, delegate.clone(), proxy_type.clone(), delay)?;
```
Comment (xlc) [inline frame/proxy/src/lib.rs:231]: better to emit event within `add_proxy_delegate` so we have still have events when new proxy is added via other pallets

**Review by thiolliere — APPROVED** [review]
code is good to me,
maybe this could also be achieve by looking at successful transaction of add_proxy_delegate.

**Comment by shaunxw** [comment]
pls review :]

**Review by bkchr — APPROVED** [review]

**Comment by bkchr** [comment]
@shaunxw please run cargo fmt to make the CI happy :P

**Comment by shaunxw** [comment]
it seems CI is happy with format now

**Comment by thiolliere** [comment]
bot merge

**Pull request merged**